### ISSUE TITLE
Using loaded event for Progress element in RadListView produce error in console but the list is correctly built on Android emulator

### ISSUE THREAD

**Issue by jgerryl** [telerik/nativescript-ui-feedback#669]
> Please take a minute to read our [NativeScript Code of Conduct](https://github.com/NativeScript/codeofconduct) before proceeding with posting issues or discussing. The purpose of this guide is to make communication and cooperation within our forums a pleasure for you and the other members.

### Did you verify this is a real problem by searching the [NativeScript Forum](https://discourse.nativescript.org/)?
Yes

### Tell us about the problem
I'm trying to use loaded event for Progress element to modify the progress bar height. This Progress element will be displayed for each item in a RadListView. It produces error in console but the list is correctly built on Android emulator.

I tried the following:
- do the same but with only 1 item in RadListView in my app -> list correctly built but still produce error.
- do the same using RadListView demo app -> list correctly built but still produce error.
- do the same but with ListView -> works fine.
- modify Progress element's height using loaded event in a view but outside of the RadListView -> works fine.

Here's the result using RadListView demo app:
error in console: 
JS: ERROR TypeError: Cannot read property 'image' of undefined
JS: ERROR CONTEXT [object Object]
(the same error is repeated for all items in the list)

but property 'image' is displayed correctly in view:
![image](https://user-images.githubusercontent.com/38002122/40289844-f6f2275a-5ce4-11e8-9a47-71758c2ceeae.png)
_

### Which platform(s) does your issue occur on?
Currently Android

### Please provide the following version numbers that your issue occurs with:
my app :
- **Progress NativeScript UI version:** nativescript-ui-listview": "^3.5.6"
- **CLI:** 4.1.0-2018-05-10-11458
- **Cross-platform modules:** "tns-core-modules": "~3.4.0"
- **Runtime(s):** "tns-android": "4.0.1"

RadListView demo app:
- **Progress NativeScript UI version:** nativescript-ui-listview": "^3.5.0"
- **CLI:** 4.1.0-2018-05-10-11458
- **Cross-platform modules:** "tns-core-modules": "~3.4.0"
- **Runtime(s):** "tns-android": "3.4.0"

### Please tell us how to recreate the issue in as much detail as possible.
1. Add Progress element in a RadListView.
2. Use loaded event for Progress element to modify the progress bar height.
3. Start Application and see the console. 

Any help would be really appreciated. Thank you.

### Is there code involved? If so, please share the minimal amount of code needed to recreate the problem.
View on demo app:

> <GridLayout tkExampleTitle tkToggleNavButton>
    <RadListView [items]="dataItems" loadOnDemandMode="Auto" (loadMoreDataRequested)="onLoadMoreItemsRequested($event)">
        <ng-template tkListItemTemplate let-item="item">
            <StackLayout class="itemTemplateStackLayout" orientation="vertical">
                <StackLayout class="innerOtemTemplateStackLayout" orientation="horizontal">
                    <Image *tkIfIOS [src]="item.image" stretch="aspectFit" width="100"></Image>
                    <FrescoDrawee *tkIfAndroid height="100" width="100" [imageUri]="item.image"></FrescoDrawee>
                    <StackLayout class="labelsStackLayout" orientation="vertical">
                        <Label class="labelName" [text]="item.name"></Label>
                        <Label class="labelTitle" [text]="item.title" textWrap="true"></Label>
                        <Label class="labelText" [text]="item.text" textWrap="true"></Label>
                    </StackLayout>
                </StackLayout>
                <Progress value="50" maxValue="100" (loaded)="onProgressLoaded($event)"></Progress>
            </StackLayout>
        </ng-template>
        <ListViewLinearLayout *tkIfIOS tkListViewLayout itemHeight="120"></ListViewLinearLayout>
    </RadListView>
</GridLayout>

==================
The code for modifying progress bar height in onProgressLoaded():

> onProgressLoaded(args) {
        let progress = <Progress>args.object;
        if (isAndroid) {
            progress.android.setScaleY(6);  //  progress.android === android.widget.ProgressBar
        } else if (isIOS) {
            let transform = CGAffineTransformMakeScale(1.0, 6.0);
            progress.ios.transform = transform; // progress.ios === UIProgressView
        }
    }

**Comment by tsonevn** [comment]
HI @jgerryl,
After building the sample project I was able to recreate the error mentioned above, however, the RadListView and the Progress components, which are used in the ListView items were correctly rendered on iOS.  Regarding that can you provide more info about the issue, which you are facing the displaying the Progress component.

**Comment by Ross-Jo** [comment]
+1

**Comment by jgerryl** [comment]
Hi @tsonevn,
Thank you for your reply. I have no problem rendering in Android either. The progress component and the RadListView are correctly displayed on the emulator. The problem is, while it's correctly rendered, it produced errors on the console.
Here's the error as I've written above:
JS: ERROR TypeError: Cannot read property 'image' of undefined
JS: ERROR CONTEXT [object Object]
(the same error is repeated for all items in the list)

Regarding your request, what other info do you need? 
Thanks.

**Comment by tsonevn** [comment]
Hi @jgerryl,
We investigated further the case, and the issue seems to be related to the Image while loading its source from the Resource folder. We logged this issue in NativeScript [modules repository](https://github.com/NativeScript/NativeScript/issues/5907), and you can keep track on it for further info.

**Comment by jgerryl** [comment]
Hi @tsonevn,
I'm sorry I didn't mention this before. This issue isn't limited only for image. The error mentions property 'image' because it happens to be the first property that is used in RadListView. I tried to delete 'image' to see what would happen but the error just changed to:
JS: ERROR TypeError: Cannot read property 'name' of undefined
JS: ERROR CONTEXT [object Object]
(property 'name' is now the first property as the 'image' property is deleted)

Property 'name' is just a string. So I don't think it's related to image loading issue. I really appreciate your help so far. Thanks.

**Comment by shabib3** [comment]
+1

**Comment by shabib3** [comment]
@jgerryl - the issue is happening also on real device or only emulator?

**Comment by tsonevn** [comment]
HI all,
I tested this case with our [sample project ](https://github.com/telerik/nativescript-ui-samples-angular/tree/master/listview/app/examples/load-on-demand) while using the latest `nativescript-ui-listview` version `3.5.9` and `tns-core-modules` version `4.1.0` and the issue is no longer reproducible.

**Comment by tsonevn** [comment]
Hi @racknoris,
Can you provide a sample project, which can be used for recreating the issue?

**Comment by racknoris** [comment]
@tsonevn Today I tried to recreate the problem- I have an NS Angular project with RadListView based on the multiple templates SDK sample. Currently I have 7 `tkTemplateKey` and 1 `tkListItemTemplate`, and a header. It seemed to have happened in the emulator when I changed the HTML and the emulator wasn't in focus when NativeScript refreshed the application- but I can't can't find any other pattern.
If I do, i'll make a reduced sample. Also, the first screen of the application contains this list, and it seem to happen in one of the first few times after emulator start.

Meanwhile, the template that returns the error is in the following code: specifically the android runtime complains about `Cannot read property 'info_array' of undefined`

```

        <ng-template tkTemplateKey="DAILY_SCHEDULE" let-item="item" let-i="index">

            <GridLayout rows="*" columns="*">          
            
                <StackLayout *ngIf="i == 0" backgroundColor="#4292F1" width="100%" height="55" class="vert-top"></StackLayout>                 

                <CardView [backgroundColor]="cardBackgroundColor" [margin]="cardRegularMargin" [elevation]="cardElevation" [radius]="cardRadious">            
                    <GridLayout rows="auto, auto, auto, auto, auto" columns="auto, *, auto, auto" class="feed-card-schedule-inner-cont">
                        <StackLayout row="0" col="3" text="" class="feed-card-schedule-generic-side-gap"></StackLayout>
                        
                        <Image row="0" col="2" src="~/images/avatar_pic_edit.png" class="feed-item-type-pic vert-cntr horiz-cntr" stretch="aspectFill"></Image>
                        
                        <Label row="0" col="0" colSpan="2" [text]="item.property_1" class="feed-item-title-text feed-item-schedule-title-gap feed-item-gap-from-type horiz-right vert-cntr"></Label>
                        

                        <StackLayout row="1" col="0" colSpan="4" orientation="vertical" class="horiz-right" width="100%">
                            <ng-template ngFor let-infoItem [ngForOf]="item.info_array" let-i="index">
                                <GridLayout rows="auto, auto, auto" columns="*, auto, auto, auto" class="feed-schecule-info-item-cont horiz-right">

                                    <Label row="0" col="3" text="&#xf111;" class="feed-schedule-dot feed-schedule-info-indent font-awesome horiz-right vert-cntr"></Label>

                                    <Label row="0" col="0" colSpan="3" [text]="infoItem.info_item_str_1" class="feed-item-title-text horiz-right vert-cntr"></Label>

                                    <Label row="1" col="0" colSpan="3" textWrap="true" [text]="infoItem.info_item_str_2" [ngClass]="{'feed-item-course-name-text feed-schedule-info-course-name-gap horiz-right vert-cntr': true, 'txt-right': isIOS}"></Label>

                                    <Image row="2" col="2" src="~/images/avatar_pic_edit.png" class="feed-schedule-info-pic vert-cntr horiz-right" stretch="aspectFit"></Image>
                                    <Label row="2" col="1" [text]="infoItem.info_item_str_3" [ngClass]="{'feed-schedule-info-teacher feed-schedule-info-bottom-segment-gap horiz-right vert-cntr': true, 'txt-right': isIOS}"></Label>
                                    <Label row="2" col="0" [text]="infoItem.info_item_str_4" [ngClass]="{'feed-schedule-info-duration horiz-right vert-cntr': true, 'txt-right': isIOS}"></Label>

                                </GridLayout>
                            </ng-template>
                        </StackLayout>

                    </GridLayout>
                </CardView>            
            </GridLayout>
        </ng-template>
```

values for cardView are:
```
        this.cardBackgroundColor = "#fff";
        this.cardFirstMargin = "0 16 22 16";
        this.cardRegularMargin = "0 16 22 16";
        this.cardElevation = 17;
        this.cardRadious = this.isIOS ? 6 : 15;
```

**Comment by racknoris** [comment]
Also, as a temporary workaround, how much will it affect cell recycling if i `*ngIf="item"` at the first `GridLayout` of my templates?

**Comment by keerl** [comment]
I am also experiencing this issue, it is very frustrating. I believe it only happens when I empty the ObservableArray completely and add new content (arrayInstance.splice(0)) then after that when it tries to load more content it will just blow up on Android sometimes (randomly).

**Comment by tsonevn** [comment]
Hi @keerl,
Can you send us the project or sample one, which demonstrates the issue and can be used for debugging?

**Comment by keerl** [comment]
Sorry for the delay, this bug has been a nightmare. I keep thinking I fix it when it doesn't happen for a while, but then it will come back randomly. I thought a complete project refactoring might fix the problem, but it didn't. 

I created a sample project that has the problem. [https://github.com/keerl/radlistview-bug](https://github.com/keerl/radlistview-bug). 

I also recorded a video of the problem happening. [https://www.youtube.com/watch?v=_EGvw8REek8](https://www.youtube.com/watch?v=_EGvw8REek8). The problem is very random, sometimes it will never trigger (which makes me think I fixed it), then randomly hours later start happening again. Only happens on Android.
